reactor-core/src/main/java/reactor/core/publisher/FluxPublishOn.java

```diff
@@ -501,6 +450,52 @@ void runBackfused() {
 			}
 		}
 
+		void runOneByOne() {
+			int missed = 1;
+
+			final Subscriber<? super T> a = actual;
+
+			long e = produced;
+			for (; ; ) {
+				long r = this.requested;
```

Comment:
shouldn't this be `this.requested & Long.MAX_VALUE` ?

Reply:
Yes, it needs to cast requested here if the drain is called before the request